src/feat/feature-common-inl.h

```diff
@@ -25,24 +25,62 @@
 namespace kaldi {
 
 template <class F>
+void OfflineFeatureTpl<F>::ComputeFeatures(
+    const VectorBase<BaseFloat> &wave,
+    BaseFloat sample_freq,
+    BaseFloat vtln_warp,
+    Matrix<BaseFloat> *output) {
+  KALDI_ASSERT(output != NULL);
+  BaseFloat new_sample_freq = computer_.GetFrameOptions().samp_freq;
+  if (sample_freq == new_sample_freq)
+    Compute(wave, vtln_warp, output);
+  else {
+    if (new_sample_freq < sample_freq) {
+      if (! computer_.GetFrameOptions().allow_downsample)
+        KALDI_ERR << "Waveform and config sample Frequency mismatch: "
+                  << sample_freq << " .vs " << new_sample_freq
+                  << "--allow_downsample should be true to allow "
+                  << " downsampling the waveform.";
+
+      // Downsample the waveform.
```

Comment:
OK, so I guess this is where you'd call the 'real' downsample function.  
Declare it in resample.h, and maybe make it something like:
```
/// Downsample a waveform.  This is a convenience wrapper for the 
/// class 'LinearResample'.
void DownsampleWaveform(BaseFloat orig_freq, const VectorBase<BaseFloat> &wave,
                BaseFloat new_freq, Vector<BaseFloat> *new_wave);
```
Regarding the filter-cutoff, please find out what sox uses and use that rule to find it;
and document it in the declaration of the function.  For 'num_zeros', maybe use 10.